Source/SwiftLintFramework/Rules/FileNameRule.swift

```diff
@@ -33,11 +38,21 @@ public struct FileNameRule: ConfigurationProviderRule, OptInRule {
             return []
         }
 
+        let prefixRegex = regex("\\A\(configuration.prefixPattern)")
+        let suffixRegex = regex("\(configuration.suffixPattern)\\z")
+
         var typeInFileName = fileName.components(separatedBy: CharacterSet(charactersIn: "+.")).first ?? fileName
 
-        let extensionSuffix = "Extension"
-        if typeInFileName.hasSuffix(extensionSuffix) {
-            typeInFileName = String(typeInFileName.dropLast(extensionSuffix.count))
+        if let prefixMatch = prefixRegex.firstMatch(in: typeInFileName, options: [], range: typeInFileName.fullNSRange) {
+            if let range = typeInFileName.nsrangeToIndexRange(prefixMatch.range) {
```

Comment:
small nit: I'd combine the two `if let`s

Reply:
Fixed.